h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -40,6 +38,45 @@ public GLMModel(Key selfKey, GLMParameters parms, GLM job, double [] ymu, double
     _nullDOF = nobs - (parms._intercept?1:0);
   }
 
+  public void initScoringInfo(GLMParameters parms, int nclasses) {
+    scoringInfo = ScoringInfo.prependScoringInfo(null, scoringInfo);
+    scoringInfo = new ScoringInfo[1];
+    scoringInfo[0].is_classification = nclasses > 1;
+    scoringInfo[0].validation = parms.valid() != null;
+    scoringInfo[0].time_stamp_ms = _output._start_time;
+  }
+
+  public ScoreKeeper[] scoreKeepers() {
+    int size = scoringInfo==null?0:scoringInfo.length;
+    ScoreKeeper[] sk = new ScoreKeeper[size];
+    for (int i=0;i<size;++i) {
+      sk[i] = scoringInfo[i].cross_validation ? scoringInfo[i].scored_xval
```

Comment:
I would recommend using parentheses when combining multiple ternary operators, overall, the formatting of the code is not ideal and its hard to read it

Reply:
I just break it out into if elseif else statements to make it clear.